desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -819,6 +833,75 @@ public void actionPerformed(ActionEvent e) {
         return panel;
     }
 
+    private ActionListener execute(final AnalysisJobBuilder analysisJobBuilder) {
+        return new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                applyPropertyValues();
+
+                if (analysisJobBuilder.getAnalyzerComponentBuilders().isEmpty()) {
+                    // Present choices to user to write file somewhere,
+                    // and then run a copy of the job based on that.
+                    ExecuteJobWithoutAnalyzersDialog executeJobWithoutAnalyzersPanel = new ExecuteJobWithoutAnalyzersDialog(
+                            _dcModule, getWindowContext(), analysisJobBuilder, _userPreferences);
+                    executeJobWithoutAnalyzersPanel.open();
+                    return;
+                }
+
+                final RunAnalysisActionListener runAnalysis = new RunAnalysisActionListener(_dcModule,
+                        analysisJobBuilder);
+                runAnalysis.run();
+            }
+        };
+    }
+
+    private ActionListener executeSingleThreaded() {
+        return new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final DataCleanerConfigurationImpl configuration = new DataCleanerConfigurationImpl(_configuration)
+                        .withEnvironment(new DataCleanerEnvironmentImpl(_configuration.getEnvironment())
+                                .withTaskRunner(new SingleThreadedTaskRunner()));
+
+                final AnalysisJob jobCopy = _analysisJobBuilder.toAnalysisJob(false);
+                final AnalysisJobBuilder jobBuilderCopy = new AnalysisJobBuilder(configuration, jobCopy);
+
+                execute(jobBuilderCopy).actionPerformed(e);
+            }
+        };
+    }
+
+    private ActionListener executePreview() {
+        return new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final String maxRowsString = JOptionPane.showInputDialog("How many records do you want to process?",
+                        "100");
+                final Number maxRows = ConvertToNumberTransformer.transformValue(maxRowsString);
+                if (maxRows == null) {
```

Comment:
Right ... Garbage would become null, but -1 would in deed become -1, so we should check for that.
Feel free to take over the branch since I'm going out for a few weeks ;-)